plutus-playground-client/src/Data/Array/Extra.purs

```diff
@@ -0,0 +1,17 @@
+module Data.Array.Extra
+  ( move
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Maybe (fromMaybe)
+
+move :: forall a. Show a => Int -> Int -> Array a -> Array a
+move source destination before
+ | destination == source = before
+ | otherwise = fromMaybe before do
+     x <- Array.index before source
```

Comment:
just because I can't think of anything else, maybe `i` is better than `x` for an index 😈

Reply:
`x` isn't the index, it's the element in the array.